### PR TITLE
Don't download/process RNA-Seq samples from GEO

### PR DESCRIPTION
**Pull request by kurtwheeler**
## Issue Number

#966 
#967 
#968 

## Purpose/Implementation Notes

The goal of this PR is to prevent samples from making it through our system and still being mislabeled. Additionally this will prevent salmon from being run on samples that came from GEO.

## Types of changes

- Bugfix (non-breaking change which fixes an issue)

## Functional tests

I reran the GEO surveyor for GSE6911 because as I mentioned here, https://github.com/AlexsLemonade/refinebio/issues/963#issuecomment-451249805, it didn't get labelled properly. With these changes, it works now!

I think everything else is well covered by unit tests.

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: common/data_refinery_common/models/models.py [hunk 1]
```diff
@@ -749,6 +749,12 @@ def needs_downloading(self, pipeline_applied=None) -> bool:
         # has been processed and doesn't need to be processed again.
         return successful_processor_jobs.count() == 0
 
+    def is_CEL_file(self) -> bool:
+        """Return true if original_file is a CEL file."""
+        upper_name = self.source_filename.upper()
+        return (len(upper_name) > 4 and upper_name[-4:] == ".CEL") \
+            or (len(upper_name) > 7 and upper_name[-7:] == ".CEL.GZ")
```
Comment (Miserlou) [inline common/data_refinery_common/models/models.py]: This I'm not sure about - is a CEL.GZ really a CEL file? Isn't it a gzip file?

File: foreman/data_refinery_foreman/surveyor/geo.py [hunk 2]
```diff
@@ -359,27 +348,57 @@ def create_experiment_and_samples_from_api(self, experiment_accession_code) -> (
...
+                # It's okay to survey RNA-Seq samples from GEO, but we
+                # don't actually want to download/process any RNA-Seq
+                # data unless it comes from SRA.
+                if sample_object.technology != 'RNA-SEQ':
+                    created_samples.append(sample_object)
+
+                # Now that we've determined the technology at the
+                # sample level, we can set it at the experiment level,
+                # just gotta make sure to only do it once.
+                # XXX: Do we want to remove this field? It's not going
```
Comment (Miserlou) [inline foreman/data_refinery_foreman/surveyor/geo.py]: If we're touching this code, can we address this concern now or remove the triple X? :vin_diesel:

File: workers/data_refinery_workers/downloaders/utils.py [hunk 3]
```diff
@@ -165,7 +165,7 @@ def delete_if_blacklisted(original_file: OriginalFile) -> OriginalFile:
         original_file.save()
         return None
 
-    return OriginalFile
+    return original_file
```
Comment (Miserlou) [inline workers/data_refinery_workers/downloaders/utils.py]: Wow, good catch :fish:

File: workers/run_tests.sh [hunk 4]
```diff
@@ -479,6 +479,7 @@ for image in ${worker_images[*]}; do
                --env AWS_ACCESS_KEY_ID \
                --env AWS_SECRET_ACCESS_KEY \
                --volume $volume_directory:/home/user/data_store \
+               --memory=5G \
```
Comment (Miserlou) [inline workers/run_tests.sh]: TIL 💡

**Comment by kurtwheeler** [comment]
Ok, I think I addressed your concern @Miserlou

**Pull request merged**